src/OSInfo.cpp

```diff
@@ -23,6 +23,20 @@
 #include "Version.h"
 
 #if defined(Q_OS_WIN)
+// regString converts a wchar_t string of size to a
+// QString. If the string contains a NUL value, that
+// NUL value will terminate the string.
+static QString regString(wchar_t *string, int size) {
+	if (size <= 0) {
```

Comment:
Is that just being cautious w.r.t to negative values and ==0 is thrown in there because we can or do wcsnlen or fromWCharArray have issues I missed with size == 0 I missed?